app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/GroupsFragment.java

```diff
@@ -0,0 +1,255 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+public class GroupsFragment extends Fragment {
+    private static final int[] GROUP_SEARCH_TITLE_ID = {
+        R.string.search_group_name, R.string.search_group_tag
+    };
+    private static final int NAME_SEARCH_MODE = 0;
+    private static final String TAG_REGEX = "^[a-zA-Z0-9,]*$";
+    private static final String INTERROGATION_SIGN = "?";
+
+    private FragmentGroupsBinding binding;
+    private List<Group> groups;
+    private CommunityCommunication communication;
+    private int selectedSearchMode;
+    private TextInputLayout inputGroupSearch;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        binding = FragmentGroupsBinding.inflate(inflater, container, false);
+        communication = (CommunityCommunication) getActivity();
+        inputGroupSearch = binding.inputGroupSearch;
+
+        setTypeSearch();
+        addSearchButtonListener();
+
+        return binding.getRoot();
+    }
+
+    /**
+     * Add the listeners to the search button.
+     */
+    private void addSearchButtonListener() {
+        binding.btnGroupSearch.setOnClickListener(v -> {
+            groups = Objects.requireNonNull(communication).getAllGroups();
+            String inputText = Objects.requireNonNull(inputGroupSearch.getEditText()).getText().toString()
+                .toLowerCase(Locale.getDefault());
+            boolean isCorrect = getGroups(inputText);
+
+            if (isCorrect) {
+                binding.groupInfoLayout.showGroups(groups);
+                addGroupsViewListeners();
+            }
+        });
+    }
+
+    /**
+     * Add the listeners to the groups view.
+     */
+    private void addGroupsViewListeners() {
+        List<CircularEntryView> views = binding.groupInfoLayout.getGroupComponents();
+
+        for (CircularEntryView circularEntryView : views) {
+            circularEntryView.setLongClickable(true);
+            circularEntryView.setOnLongClickListener(v1 -> {
+                MaterialAlertDialogBuilder dialog = createDeleteGroupDialog(circularEntryView);
+                dialog.show();
+                return true;
+            });
+        }
+    }
+
+    /**
+     * Create a dialog to delete the group.
+     * @param circularEntryView The entry to which the dialog is associated to
+     * @return The dialog that is associated with the entry
+     */
+    private MaterialAlertDialogBuilder createDeleteGroupDialog(CircularEntryView circularEntryView) {
+        MaterialAlertDialogBuilder dialog = new MaterialAlertDialogBuilder(Objects.requireNonNull(getContext()),
+            R.style.AlertDialogTheme);
+        Group group = (Group) circularEntryView.getObject();
+        dialog.setTitle(getString(R.string.delete_group_title) + " " + group.getName() + INTERROGATION_SIGN);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String " " appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5378994)